lib/SILOptimizer/Mandatory/TFCanonicalizeCFG.cpp

```diff
@@ -657,10 +657,31 @@ SILBasicBlock *SingleExitLoopTransformer::createNewExitBlockWithDemux(
 }
 
 bool SingleExitLoopTransformer::transform() {
-  // Return if the loop is already in the required form.
+  // Return if the loop is already in the required form and
+  // if the header has no side-effects.
   if (loop->getExitBlock() && loop->getExitingBlock() &&
-      loop->getExitingBlock() == loop->getHeader()) {
-    return false;
+      loop->getExitingBlock() == loop->getHeader())  {
+    constexpr char opSendToHost[] = "tfc.SendToHost";
```

Comment:
I don't see a need for `constexpr` because they are local and are already constant.